core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineMysql.java

```diff
@@ -444,4 +446,9 @@ public TimestampTZColumn parseTimestampTZColumn(ResultSet resultSet, String colu
       getTimeTypeStrategy() {
     return timeTypeEngine;
   }
+
+  @Override
+  public Map<String, String> getConnectionProperties() {
+    return Collections.singletonMap("useCursorFetch", "true");
```

Comment:
Also, for MySQL, we need to set `useCursorFetch=true` in the connection properties.